web/src/main/java/com/marklogic/hub/web/service/FlowJobService.java

```diff
@@ -166,11 +179,33 @@ private FlowJobs retrieveJobsByFlowName(String flowName) {
                 latestJob.output = s.get("stepOutput"); //last step output ?
             });
         }
+
+        if (!staleStateJobIds.isEmpty()) {
+            updateJobStaleStates(staleStateJobIds);
+        }
         this.release();
 
         return flowJobs;
     }
 
+    /**
+     * Set the jobStatus as Failed and timeEnded due to server failure during flow running
+     * @param staleStateJobIds
+     */
+    private void updateJobStaleStates(List<String> staleStateJobIds) {
+        RequestParameters params = new RequestParameters();
+        for (String jobId : staleStateJobIds) {
+            params.add("jobid", jobId);
+            params.add("status", JobStatus.FAILED.toString());
+
+            try {
+                this.getServices().post(params, new StringHandle("{}").withFormat(Format.JSON));
+            } catch (Exception e) {
+                logger.error(e.getMessage());
```

Comment:
Can we log the job-id explicitly that was associated with the error in addition to the exception?